File: programmers/etc/level2/멀쩡한_사각형.py
def gcd(a,b):
    while b:
        a, b = b, a%b
    return a

def solution(w,h):
    tot = w*h
    g = gcd(w,h)
    w, h = w//g, h//g
    cut = w+h-1
    return tot - g*cut


w,h = 8,12
print(solution(w,h)) # 80

File: Java/programmers/etc/level2/멀쩡한사각형.java
package CodeTest.Java.programmers.etc.level2;

public class 멀쩡한사각형 {
    public long solution(int w, int h) {
        long total = (long) w*h;
        int g = gcd(w,h);
        w /= g; h /= g;
        int cut = w+h-1;
        return total - (long) g*cut;
    }

    private static int gcd(int w, int h) {
        while (h != 0) {
            int tmp = w;
            w = h; h = tmp%h;
        }
        return w;
    }
}
